src/libs/tools/include/merging/mergingkdb.hpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ */
+
+#ifndef KDBMERGE_HPP_
+#define KDBMERGE_HPP_
+
+#include <kdb.hpp>
+#include <merging/threewaymerge.hpp>
+
+namespace kdb
+{
+
+namespace tools
+{
+
+namespace merging
+{
+
+/**
+ * Provides a merging wrapper around a KDB instance. The wrapper allows to pass
+ * a three way merger instance that is used to resolve conflicts during KDB set.
+ */
+class MergingKDB {
```

Comment:
> why not derive from KDB?

I was uncertain, but maybe inheritance is a cleaner solution in this case. However, the `set` interface should stay different from the base variant because it has a slightly different behaviour (due to the get the passed KeySet is updated). Do you agree?